content/post/051-new-ipfs-release-process.md

```diff
@@ -0,0 +1,209 @@
+would have been caught by better release testing and released faster with an
+improved release process.
+
+* 0.4.21 had two performance regressions in bitswap:
+  * a throughput regression that should have been caught by regression testing
+    (now tested) but almost certainly would have been noticed by downstream
+    users given a longer release-testing process.
+  * a CPU utilization regression that only shows up with >10000 peers. Something
+    that only really shows up in certain production systems under heavy load.
+* 0.4.20 had a regression where adding multiple independent files in the same
+  add command didn't work ([#6254](https://github.com/ipfs/go-ipfs/pull/6255)).
+  A regression test has since been added but this _also_ would have been caught
+  with better cross-application testing.
+* 0.4.19 had multiple regressions:
+  * a regression in the docker container (introduced by
+    [#6040](https://github.com/ipfs/go-ipfs/pull/6040)) that would have been
+    caught by testing the go-ipfs docker image in more production environments.
+  * a CPU utilization regression in bitswap only seen under very high load. This
+    would have been caught by testing under production loads.
+  * panics in the DHT and QUIC modules that only show up under heavy load.
+
+## Actions
+  
+To prevent this pattern from continuing, we've taken several actions:
+
+1. To address the stability issues, we've introduced a new release process that
+   involves extensively testing releases in a wide variety of production
+   environments.
+2. To address the issue of slow releases, we've introduced a 6 week release
+   cycle.
+3. To address the issue of slow bug fixes, we've switched to semver and
+   introduced patch releases. The first patch release will be 0.4.22 and the
+   next _feature_ release will be 0.5.0.
+   
+At the same time, we've introduced a feature release _freeze_ until we can
+finish building out our testing infrastructure (a topic for another blog post).
+
+XXX: Should we write that one first? It's awkward to say "we're introducing a
+new release cadence" and then just say "but we're not using it yet :p".
+
+### Early Testers Program
+
+The early testers program allows groups using go-ipfs in production to
+self-volunteer to help with testing `go-ipfs` release candidates in development
+and production environments. While we invite the _entire_ community to help test
+releases, members of the early testers program participate directly and actively
+in every release.
+
+Early testers will deploy release candidates to dev and prod environments,
+giving us quick feedback on any regressions or performance changes they notice.
+This means we can get some quick feedback from heavy users before we cut the
+official release.
+
+This program currently includes:
+
+- Infura
+- Textile
+- Pinata
+- RTrade
+- QRI
+- Siderus
+
+XXX: should get signoffs to make sure publishing this list in a blog post is fine.
```

Comment:
?

Reply:
probably worth pining on the original invite something along the lines of "we're sending out a short announcement about the early tester program and your participation later this week - LMK if you'd prefer not to be mentioned/listed in the early testers list in that post" - I don't imagine any of them will want to opt out...

Reply:
Y'all. Any objections to being thanked for signing up for the Early Testers program in a blog post?

- [ ] Infura (@MichaelMure)
- [ ] Textile (@sanderpick)
- [ ] Pinata (@obo20)
- [ ] RTrade (@postables)
- [ ] QRI (@b5)
- [ ] Siderus (@koalalorenzo)

Reply:
No problems here!

Reply:
SGTM. Thanks, @Stebalien.

Reply:
delighted. Thanks!

Reply:
Sounds great, thanks!

Reply:
All fine by me :D